File: src/actions/javascript/rekeyAction.ts
import { KmsClient } from "cloudproof_js";
import { BACKEND_URL } from "./backendConfig";
//
// Perform a rekey operation
//
export const rekeyAccessPolicy = async (kmsToken: string, masterSecretKeyUId: string): Promise<string[]> => {
  const client = new KmsClient(BACKEND_URL, kmsToken);
  return await client.rekeyCoverCryptAccessPolicy(masterSecretKeyUId, "country::Germany");
};

File: public/actions/java/rekeyAction.java
//
// REKEY ACTION
//
public static void rekeyAccessPolicy(String kmsServerUrl, String apiKey, String privateMasterKeyUniqueIdentifier) {
  KmsClient kmsClient = new KmsClient(kmsServerUrl, apiKey);
  kmsClient.rekeyCoverCryptAccessPolicy(privateMasterKeyUniqueIdentifier,"Country::Germany");
}
